### PR TITLE
Refactor and simplify GitPushDialog

### PR DESCRIPTION
**Pull request by mkondratek**

### CODE REVIEW

File: CHANGE-NOTES.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## v3.2.2
 - Improve CI build stability.
+- Enhance the (Force) Push dialog with the default help action, the push tags option and the title indicating the repository.
```
Comment (PawelLipski) [inline CHANGE-NOTES.md:5]: Wow indeed

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java [hunk 2]
```diff
@@ -58,128 +25,22 @@ public GitPushDialog(
...
 
     // Note: since the class is final, `this` is already @Initialized at this point.
 
-    dialog.updateOkActions();
     setOKButtonText(getPushActionName());
-    setTitle("Push Commits");
     init();
   }
 
-  private class MyVcsPushDialog extends VcsPushDialog {
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java:81]: Also... can `com.virtuslab.archunit.ClassStructureTestSuite#inner_classes_should_not_be_instantiated_from_constructor_of_enclosing_class` be updated now?

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java [hunk 3]
```diff
@@ -190,130 +51,19 @@ protected void doOKAction() {
...
-          result.set(PrePushHandler.Result.ABORT);
-        }
-      }
-    }.queue();
-
-    PrePushHandler.Result resultValue = result.get();
-    assert resultValue != null : "Result value is null";
-    return resultValue;
-  }
-
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java:316]: :benissimo: code removal mmmmmm

**Review by PawelLipski — APPROVED** [review]

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java (hunk 2 above)
Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java:81]: sadly no 😢 there is still this line:
```java
this.pushAction = new PushSwingAction()
```

**Pull request merged**